azurerm/internal/services/compute/virtual_machine_scale_set.go

```diff
@@ -1297,3 +1299,185 @@ func FlattenVirtualMachineScaleSetAutomaticRepairsPolicy(input *compute.Automati
+			ProvisionAfterExtensions: utils.ExpandStringSlice(extensionRaw["provision_after_extensions"].([]interface{})),
+		}
+
+		if forceUpdateTag := extensionRaw["force_update_tag"]; forceUpdateTag != nil {
+			extensionProps.ForceUpdateTag = utils.String(forceUpdateTag.(string))
+		}
+
+		settings, _ := structure.ExpandJsonFromString(extensionRaw["settings"].(string))
+		extensionProps.Settings = settings
+
+		protectedSettings, _ := structure.ExpandJsonFromString(extensionRaw["protected_settings"].(string))
+		extensionProps.ProtectedSettings = protectedSettings
+
+		extension.VirtualMachineScaleSetExtensionProperties = &extensionProps
+		extensions = append(extensions, extension)
+	}
+	result.Extensions = &extensions
+
+	return result
+}
+
+func flattenVirtualMachineScaleSetExtensions(input *compute.VirtualMachineScaleSetExtensionProfile, d *schema.ResourceData) ([]map[string]interface{}, error) {
+	result := make([]map[string]interface{}, 0)
+	if input == nil {
+		return result, nil
+	}
+
+	for k, v := range *input.Extensions {
+		ext := make(map[string]interface{})
+		name := ""
+		if v.Name != nil {
+			name = *v.Name
+		}
+		ext["name"] = name
+
+		if props := v.VirtualMachineScaleSetExtensionProperties; props != nil {
+			extPublisher := ""
+			if props.Publisher != nil {
+				extPublisher = *props.Publisher
+			}
+			ext["publisher"] = extPublisher
+
+			extType := ""
+			if props.Type != nil {
+				extType = *props.Type
+			}
+			ext["type"] = extType
+
+			extTypeVersion := ""
+			if props.TypeHandlerVersion != nil {
+				extTypeVersion = *props.TypeHandlerVersion
+			}
+			ext["type_handler_version"] = extTypeVersion
+
+			autoUpgradeMinorVersion := false
+			if props.AutoUpgradeMinorVersion != nil {
+				autoUpgradeMinorVersion = *props.AutoUpgradeMinorVersion
+			}
+			ext["auto_upgrade_minor_version"] = autoUpgradeMinorVersion
+
+			forceUpdateTag := ""
+			if props.ForceUpdateTag != nil {
+				forceUpdateTag = *props.ForceUpdateTag
+			}
+			ext["force_update_tag"] = forceUpdateTag
+
+			provisionAfterExtension := make([]interface{}, 0)
+			if props.ProvisionAfterExtensions != nil {
+				provisionAfterExtension = utils.FlattenStringSlice(props.ProvisionAfterExtensions)
+			}
+			ext["provision_after_extensions"] = provisionAfterExtension
+
+			extSettings := ""
+			if props.Settings != nil {
+				extSettingsRaw, err := structure.FlattenJsonToString(props.Settings.(map[string]interface{}))
+				if err != nil {
+					return nil, err
+				}
+				extSettings = extSettingsRaw
+			}
+			ext["settings"] = extSettings
+
+			// protected_settings isn't returned, so we attempt to get it from config otherwise set to empty string
+			if protectedSettings, ok := d.GetOk(fmt.Sprintf("extension.%d.protected_settings", k)); ok {
+				if protectedSettings.(string) != "" && protectedSettings.(string) != "{}" {
+					ext["protected_settings"] = protectedSettings.(string)
+				}
+			} else {
+				ext["protected_settings"] = ""
```

Comment:
this field won't be set if the existing `protected_settings` is an empty string / `{}` - can we set the existing contents to a variable and then assign this key after the if statement?